Adding command line example

jose-compu opened the pull request:
`USAGE:
    cli [FLAGS] <SECRET_KEY> <MESSAGE> [PROOF]`

Commit:
For client version just generating on stdout VRF proof based on hex secret and message input.

Complete command line client including verification also, proof parameter (optional).

Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

Commit:
For client version just generating on stdout VRF proof based on hex secret and message input.

Commit:
Complete command line client including verification also, proof parameter (optional).

girazoki commented on examples/cli.rs:
```diff
@@ -0,0 +1,75 @@
+    let beta = vrf.verify(&public_key, &pi, &message);
+
+    if proof_given {
+
```
```suggestion
```

girazoki commented on examples/cli.rs:
```diff
@@ -0,0 +1,75 @@
+    )
+    .get_matches();
+
+
```
```suggestion
```

girazoki commented on examples/cli.rs:
```diff
@@ -0,0 +1,75 @@
+gen_validator!(is_hex_ok, hex::decode);
+
+fn main() {
+
```
```suggestion
```

girazoki commented on examples/cli.rs:
```diff
@@ -0,0 +1,75 @@
+//! 3. (Optional) Convert the VRF proof to a hash (e.g. to be used as pseudo-random value)
+//! 4. Verify a VRF proof by using `verify()` function
+
+use vrf::openssl::{CipherSuite, ECVRF};
```
```suggestion
use hex;
use vrf::openssl::{CipherSuite, ECVRF};
```

girazoki commented on examples/cli.rs:
```diff
@@ -0,0 +1,75 @@
+
+use vrf::openssl::{CipherSuite, ECVRF};
+use vrf::VRF;
+use hex;
```
```suggestion
```

jose-compu committed:
Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

jose-compu committed:
Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

jose-compu committed:
Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

jose-compu committed:
Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

jose-compu committed:
Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

jose-compu committed:
Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

jose-compu commented:
Changes accepted!

girazoki commented on examples/cli.rs:
```diff
@@ -0,0 +1,72 @@
+            }
+        }
+    }
+
```
```suggestion
```

jose-compu committed:
Update examples/cli.rs

Co-authored-by: girazoki <gorka.irazoki@gmail.com>

jose-compu commented:
Changes accepted.

girazoki commented:
Hi @manfr3d and thanks for the contribution! Can you squash your commits into the latest one? We require to have a clean and verified history in the repo. Thanks in advance!

Commit:
Merge branch 'master' of github.com:manfr3d/vrf-rs

jose-compu commented:
That works is you neeed to remove the last commit? Thanks.

girazoki commented:
No, just join all the commits together into the latest

Commit:
Merge branch 'master' of github.com:manfr3d/vrf-rs

jose-compu commented:
Works now?

girazoki commented:
I squashed them for you @manfr3d. This PR is closed by a15f43744756fe254ad42d8ed0367595ea1b6b2d
